File: java/count-number-of-teams.java
class Solution {
    public int numTeams(int[] rating) {
        int res = 0;
        int n = rating.length;

        for(int i =0 ;i <n ; i++){
            int leftLess = 0;
            int leftGreater = 0;
            int rightLess = 0;
            int rightGreater = 0;

            for(int j = 0; j < i; j++){
                if(rating[j] < rating[i]){
                    leftLess++;
                }else if(rating[j] > rating[i]){
                    leftGreater++;
                }
            }

            for(int j = i+1; j < n; j++){
                if(rating[j] < rating[i]){
                    rightLess++;
                }else if(rating[j] > rating[i]){
                    rightGreater++;
                }
            }

            res += leftLess * rightGreater + leftGreater * rightLess;
        }
        
       
       return res; 
    }
}

File: cpp/count-number-of-teams.cpp
#include "math.h"
#include "vector"

using namespace std;

class Solution {
public:
  int numTeams(vector<int>& rating) {
    int res = 0;
    int n = rating.size();
    for (int i = 1; i < n - 1; i++) {
      int left_less = 0, left_more = 0, right_less = 0, right_more = 0;
      for (int j = 0; j < i; j++) {
        if (rating[j] < rating[i]) left_less++;
        if (rating[j] > rating[i]) left_more++;
      }
      for (int j = i + 1; j < n; j++) {
        if (rating[j] < rating[i]) right_less++;
        if (rating[j] > rating[i]) right_more++;
      }
      res += left_less * right_more + left_more * right_less;
    }

    return res;
  }
};

int main() {
  Solution s;

  return 0;
}